kotlinx-coroutines-core/common/src/internal/DispatchedTask.kt

```diff
@@ -120,7 +125,7 @@ internal fun <T> DispatchedTask<T>.resume(delegate: Continuation<T>, useMode: In
     val result = if (exception != null) Result.failure(exception) else Result.success(state as T)
     when (useMode) {
         MODE_ATOMIC_DEFAULT -> delegate.resumeWith(result)
```

Comment:
Makes sense to make extensions `inline` and use them there instead of hand-rolled comparison?

Reply:
I don't want to lose a nice table-switch in this bytecode. Moreover, this particular code is "safe", since it has "catch all" else branch just in case some option is missing.